compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7564,12 +7564,28 @@ private List<BLangExpression> concatSimilarKindXMLNodes(List<BLangExpression> ex
         List<BLangExpression> tempConcatExpressions = new ArrayList<>();
 
         for (BLangExpression expr : exprs) {
-            BType exprType;
-            if (expr.getKind() == NodeKind.QUERY_EXPR) {
+            boolean prevNonErrorLoggingCheck = data.nonErrorLoggingCheck;
+            data.nonErrorLoggingCheck = true;
+            int prevErrorCount = this.dlog.errorCount();
+            this.dlog.resetErrorCount();
+            this.dlog.mute();
+
+            BType exprType = checkExpr(nodeCloner.cloneNode(expr), xmlElementEnv, symTable.xmlType, data);
```

Comment:
I believe you need to increment the clone attempt on `expr`, or the node may not be cloned if has already been cloned.

```suggestion
            expr.cloneAttempt++;
            BType exprType = checkExpr(nodeCloner.cloneNode(expr), xmlElementEnv, symTable.xmlType, data);
```

Reply:
aren't we incrementing the `cloneAttempt` here for every clone?
https://github.com/ballerina-platform/ballerina-lang/blob/4175b5491ca992402a9ff93a47db3899ad7acb31/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/NodeCloner.java#L267

Reply:
My bad, `clone` is what doesn't increment.